IBF-Typhoon-model/src/typhoonmodel/utility_fun/Rainfall_data.py

```diff
@@ -62,6 +70,34 @@ def download_rainfall_nomads(Input_folder, path, Alternative_data_point):
             logger.info(f'Writing to file {filepath}')
             ds = ds.median(dim='number') #store only the median of the ensemble members
             ds.to_netcdf(filepath)
+        #zonal stats to calculate rainfall per manucipality 
+        rain_6h=rasterio.open(filepath) 
```

Comment:
The file operations should be inside a `with` block so that the file will be closed after:
```
with rasterio.open(filepath) as rain_6h:
```